### PR TITLE
Adding details to Conformance Tests using RFC 2119 standards.

### PR DESCRIPTION
**Pull request by brahmaroutu**
This PR is part of the conformance documentation. This is to provide more formal specification using RFC 2119 keywords to describe the test so that who ever is running conformance tests do not have to go through the code to understand why and what is tested.
The documentation information added here into each of the tests eventually result into a document which is currently checked in at location https://github.com/cncf/k8s-conformance/blob/master/docs/KubeConformance-1.9.md

I would like to have this PR reviewed for v1.10 as I consider it important to strengthen the conformance documents.

### CODE REVIEW

**Comment by jdumars** [comment]
/ok-to-test
Hi, please sort out your release note label when you can.  Thanks!

File: test/e2e/apps/rc.go [hunk 1]
```diff
@@ -38,6 +38,12 @@ import (
 var _ = SIGDescribe("ReplicationController", func() {
 	f := framework.NewDefaultFramework("replication-controller")
 
+	/*
+		Release : v1.9
```
Comment (soltysh) [inline test/e2e/apps/rc.go]: I don't see any value in having the release specified here. Nobody will update it and it'll just rot. The rest is ok.

**Review by soltysh — CHANGES_REQUESTED** [review]

File: test/e2e/apps/rc.go (hunk 1 above)
Comment (brahmaroutu) [reply test/e2e/apps/rc.go]: I understand, but the reason I proposed this to the conformance group is to track when a conformance test is added or when it is modified. This is a free form string and up to the reviewer to pay attention to it. All this meta data will show up in a document like https://github.com/cncf/k8s-conformance/blob/master/docs/KubeConformance-1.9.md

**Comment by brahmaroutu** [comment]
@soltysh Please review.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**